File: src/main/java/me/udnek/fnafu/mechanic/Time.java
package me.udnek.fnafu.mechanic;

import me.udnek.fnafu.utils.Resettable;

public class Time implements Resettable {

    private int time;
    private final int maxTime;

    public Time(int maxTime){
        this.maxTime = maxTime;
    }

    public void tick(){
        time++;
    }
    public boolean isEnded(){
        return time >= maxTime;
    }

    public int get(){return time;}

    @Override
    public void reset() {
        time = 0;
    }
}

File: src/main/java/me/udnek/fnafu/mechanic/Time.kt
package me.udnek.fnafu.mechanic

import me.udnek.fnafu.util.Resettable
import me.udnek.fnafu.util.Ticking

class Time(private val maxTime: Int) : Resettable, Ticking {
    var ticks = 0
        private set

    val isEnded: Boolean
        get() = ticks >= maxTime

    override fun tick() {
        ticks++
    }
    override fun reset() { ticks = 0 }
}
